glib2 is not imported on macOS: `amule.cpp: fatal error: glib.h: No such file or directory`

CMakeLists requires `glib2` only for Linux:
```
if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
	find_package (PkgConfig QUIET)
	if (PkgConfig_FOUND)
		pkg_check_modules (GLIB QUIET glib-2.0)
	endif()

	if ((BUILD_MONOLITHIC OR BUILD_DAEMON OR BUILD_REMOTEGUI) AND NOT GLIB_FOUND)
		message (FATAL_ERROR
			"glib-2.0 development headers not found, but they are "
			"required to build amule (monolithic), amuled, or "
			"amulegui — these targets call g_set_prgname() to bind "
			"the Wayland wl_app_id / X11 WM_CLASS to the .desktop "
			"filename, which needs <glib.h>. "
			"Install one of:\n"
			"  - libglib2.0-dev   (Debian / Ubuntu / Mint)\n"
			"  - glib2-devel      (Fedora / RHEL / Rocky)\n"
			"  - dev-libs/glib    (Gentoo)\n"
			"  - glib             (Arch / Manjaro)\n"
			"…then re-run cmake AFTER deleting the build directory "
			"(`rm -rf build`); pkg_check_modules only runs at "
			"configure time, so installing the package on top of a "
			"cached configure won't help. pkg-config itself is also "
			"required — install `pkg-config` if cmake didn't find it.")
	endif()
endif()
```

However the build uses it then on macOS, and expectedly it fails:
```
[ 56%] Building CXX object src/CMakeFiles/amule.dir/amule.cpp.o
cd /opt/local/var/macports/build/amule-05862ffd/work/build/src && /opt/local/bin/g++-mp-14 -DAMULE_HAVE_LIBCURL=1 -DENABLE_IP2COUNTRY -DNDEBUG -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D__WXGTK3__ -D__WXGTK__ -I/opt/local/var/macports/build/amule-05862ffd/work/build -I/opt/local/var/macports/build/amule-05862ffd/work/build/src/libs/ec/cpp -I/opt/local/libexec/boost/1.88/include -I/opt/local/var/macports/build/amule-05862ffd/work/amule-377ec8d9d56618f1059d8705ae53a70fa4ffbe87/src/include -I/opt/local/var/macports/build/amule-05862ffd/work/amule-377ec8d9d56618f1059d8705ae53a70fa4ffbe87/src/libs -I/opt/local/var/macports/build/amule-05862ffd/work/amule-377ec8d9d56618f1059d8705ae53a70fa4ffbe87/src/icons -isystem /opt/local/include -isystem /opt/local/Library/Frameworks/wxWidgets.framework/Versions/wxGTK/3.2/lib/wx/include/gtk3-unicode-3.2 -isystem /opt/local/include/upnp -pipe -I/opt/local/libexec/boost/1.88/include -Os -framework CoreFoundation -framework CoreServices -framework ApplicationServices -framework IOKit -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/libexec/boost/1.88/include -I/opt/local/include -Damule_HAVE_LIBCURL -I/opt/local/Library/Frameworks/wxWidgets.framework/Versions/wxGTK/3.2/include/wx-3.2 -D_GLIBCXX_USE_CXX11_ABI=0 -arch ppc -mmacosx-version-min=10.6 -MD -MT src/CMakeFiles/amule.dir/amule.cpp.o -MF CMakeFiles/amule.dir/amule.cpp.o.d -o CMakeFiles/amule.dir/amule.cpp.o -c /opt/local/var/macports/build/amule-05862ffd/work/amule-377ec8d9d56618f1059d8705ae53a70fa4ffbe87/src/amule.cpp
/opt/local/var/macports/build/amule-05862ffd/work/amule-377ec8d9d56618f1059d8705ae53a70fa4ffbe87/src/amule.cpp:55:10: fatal error: glib.h: No such file or directory
   55 | #include <glib.h>                       // g_set_prgname() — wl_app_id / WM_CLASS binding
      |          ^~~~~~~~
compilation terminated.
make[2]: *** [src/CMakeFiles/amule.dir/amule.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/opt/local/var/macports/build/amule-05862ffd/work/build'
make[1]: *** [src/CMakeFiles/amule.dir/all] Error 2
make[1]: Leaving directory `/opt/local/var/macports/build/amule-05862ffd/work/build'
make: *** [all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -250,14 +250,14 @@ include (cmake/bfd.cmake)
 # binding via g_set_prgname() (called from amule.cpp on the monolithic
 # / daemon side, and from amule-remote-gui.cpp on the amulegui side
 # once the remote-GUI fix lands as a separate PR). amulecmd and
-# amuleweb don't need glib. Detect it here best-effort so the
-# optional `if (GLIB_FOUND)` blocks downstream pick it up when
-# present (e.g. amulegui's MuleTrayIcon SNI backend), and promote
-# the missing-dep case to a hard configure-time error when a target
-# that depends on it is being built (#562 — the silent QUIET
-# surfaced as a confusing `glib.h: No such file` mid-compile for
-# fresh-checkout users).
-if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+# amuleweb don't need glib. Gated on wxGTK being the wx toolkit AND
+# the host not being macOS — the `#include <glib.h>` / call sit behind
+# `#if defined(__WXGTK__) && !defined(__APPLE__)` in those files;
+# macOS doesn't run Wayland and macOS apps identify via Info.plist, so
+# the call is a no-op there even when wxGTK happens to be the toolkit
+# (MacPorts wxgtk, see #641) and there's no reason to drag glib2 in.
+list (FIND wxWidgets_DEFINITIONS "__WXGTK__" _amule_wxgtk_idx)
+if (_amule_wxgtk_idx GREATER -1 AND NOT APPLE)
 	find_package (PkgConfig QUIET)
 	if (PkgConfig_FOUND)
 		pkg_check_modules (GLIB QUIET glib-2.0)
@@ -267,14 +267,16 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
 		message (FATAL_ERROR
 			"glib-2.0 development headers not found, but they are "
 			"required to build amule (monolithic), amuled, or "
-			"amulegui — these targets call g_set_prgname() to bind "
-			"the Wayland wl_app_id / X11 WM_CLASS to the .desktop "
-			"filename, which needs <glib.h>. "
-			"Install one of:\n"
+			"amulegui against wxGTK — these targets call "
+			"g_set_prgname() to bind the Wayland wl_app_id / X11 "
+			"WM_CLASS to the .desktop filename, which needs "
+			"<glib.h>. Install one of:\n"
 			"  - libglib2.0-dev   (Debian / Ubuntu / Mint)\n"
 			"  - glib2-devel      (Fedora / RHEL / Rocky)\n"
 			"  - dev-libs/glib    (Gentoo)\n"
 			"  - glib             (Arch / Manjaro)\n"
+			"  - glib2            (MacPorts)\n"
+			"  - glib             (Homebrew)\n"
 			"…then re-run cmake AFTER deleting the build directory "
 			"(`rm -rf build`); pkg_check_modules only runs at "
 			"configure time, so installing the package on top of a "
@@ -282,6 +284,7 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
 			"required — install `pkg-config` if cmake didn't find it.")
 	endif()
 endif()
+unset (_amule_wxgtk_idx)
 
 # Optional StatusNotifierItem (SNI) backend for the system-tray icon.
 # wxGTK's wxTaskBarIcon talks the legacy GtkStatusIcon API, which

--- a/src/amule-remote-gui.cpp
+++ b/src/amule-remote-gui.cpp
@@ -29,7 +29,7 @@
 #include <wx/fileconf.h>		// Needed for wxFileConfig
 #include <wx/socket.h>			// Needed for wxSocketBase
 
-#ifdef __WXGTK__
+#if defined(__WXGTK__) && !defined(__APPLE__)
 #include <glib.h>			// g_set_prgname() — wl_app_id / WM_CLASS binding
 #endif
 
@@ -260,7 +260,7 @@ bool CamuleRemoteGuiApp::OnInit()
 	amuledlg = NULL;
 	connect_timeout_timer = NULL;
 
-#ifdef __WXGTK__
+#if defined(__WXGTK__) && !defined(__APPLE__)
 	// Set the GTK program name to the canonical app id. On Wayland,
 	// GTK derives wl_app_id (xdg_toplevel.set_app_id) from
 	// g_get_prgname(); compositors match wl_app_id against the
@@ -274,6 +274,10 @@ bool CamuleRemoteGuiApp::OnInit()
 	// has in CamuleApp::OnInit; amulegui shipped without it, so on
 	// GNOME / wlroots the taskbar icon never bound to the launcher
 	// and showed the generic fallback. (#562 follow-up.)
+	// Skipped on macOS even under wxGTK (MacPorts): no Wayland or
+	// .desktop binding exists, and app identity is set via Info.plist
+	// in the .app bundle. Dropping the call lets that build skip the
+	// glib2 dep entirely (#641).
 	g_set_prgname("org.amule.aMule.gui");
 #endif
 

--- a/src/amule.cpp
+++ b/src/amule.cpp
@@ -51,7 +51,7 @@
 #include <wx/filename.h>		// Needed for wxFileName (CA bundle lookup)
 #endif
 
-#ifdef __WXGTK__
+#if defined(__WXGTK__) && !defined(__APPLE__)
 #include <glib.h>			// g_set_prgname() — wl_app_id / WM_CLASS binding
 #endif
 
@@ -441,7 +441,7 @@ bool CamuleApp::OnInit()
 	wxDebugContext::SetCheckpoint();
 #endif
 
-#ifdef __WXGTK__
+#if defined(__WXGTK__) && !defined(__APPLE__)
 	// Set the GTK program name to the canonical app id. On Wayland,
 	// GTK derives wl_app_id (xdg_toplevel.set_app_id) from
 	// g_get_prgname(); compositors match wl_app_id against the
@@ -452,6 +452,10 @@ bool CamuleApp::OnInit()
 	// On X11 the same value also feeds into WM_CLASS, matching
 	// StartupWMClass=org.amule.aMule in the .desktop file. Must run
 	// before any GTK window is created.
+	// Skipped on macOS even under wxGTK (MacPorts): no Wayland or
+	// .desktop binding exists, and app identity is set via Info.plist
+	// in the .app bundle. Dropping the call lets that build skip the
+	// glib2 dep entirely (#641).
 	g_set_prgname("org.amule.aMule");
 #endif
 

